portfolio/src/main/webapp/script.js

```diff
@@ -27,24 +27,52 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-//deprecated
-function getQuoteOld() {
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('quote-container').innerText = quote;
-  });
-  console.log(response)
-  console.log(quote)
+
+function showComments() {
+    fetch('/data').then(response => response.json()).then((commentsJson) => {
+     
+        // unordered list
+        var ul = document.createElement('ul');
+
+        // where comments go
+        document.getElementById('comments-container').appendChild(ul);
+
+        // iterate through json object and make into html list
+        for (var key in commentsJson) {
+            let li = document.createElement('li');
+            li.innerHTML = commentsJson[key];
+            ul.appendChild(li);
+        }
+        // BUG: currently appends all comments each time "show comments" button is clicked
```

Comment:
I think you'll have to clear the comments-container's children at the beginning of this function

Reply:
Thanks